tests/core/Blockchain.spec.ts

```diff
@@ -0,0 +1,227 @@
+  describe('read()', async () => {
+    it('should return transactions fetched.', async () => {
+      const blockchainClient = new Blockchain('Unused URI');
+      const mockFetchResponse = {
+        status: 200,
+        body: {
+          read: () => { return 'Ignored body.'; }
+        }
+      };
+      const mockReadableStreamReadString = JSON.stringify({
+        moreTransactions: false,
+        transactions: [{
+          anchorString: '1',
+          transactionNumber: 1,
+          transactionTime: 1,
+          transactionTimeHash: '1'
+        }]
+      });
+      spyOn(blockchainClient as any, 'fetch').and.returnValue(Promise.resolve(mockFetchResponse));
+      spyOn(ReadableStream, 'readAll').and.returnValue(Promise.resolve(mockReadableStreamReadString));
+
+      const readResult = await blockchainClient.read();
+      expect(readResult.moreTransactions).toBeFalsy();
+      expect(readResult.transactions.length).toEqual(1);
+      expect(readResult.transactions[0].transactionNumber).toEqual(1);
+    });
+
+    it('should throw SidetreeError with correct code when if invalid time hash is given.', async () => {
+      const blockchainClient = new Blockchain('Unused URI');
+      const mockFetchResponse = {
+        status: 400,
+        body: {
+          read: () => { return 'Ignored body.'; }
+        }
+      };
+      spyOn(blockchainClient as any, 'fetch').and.returnValue(Promise.resolve(mockFetchResponse));
+      spyOn(ReadableStream, 'readAll').and.returnValue(Promise.resolve(JSON.stringify({
+        code: SharedErrorCode.InvalidTransactionNumberOrTimeHash
+      })));
+
+      try {
+        await blockchainClient.read();
+      } catch (error) {
+        // Throwing error is the expected case.
+
+        if (error.code !== SharedErrorCode.InvalidTransactionNumberOrTimeHash) {
+          fail();
+        }
+
+        return;
+      }
+
+      fail();
+    });
+
+    it('should throw if error is encountered when reading transactions.', async () => {
+      const blockchainClient = new Blockchain('Unused URI');
+      const mockFetchResponse = {
+        status: 500,
+        body: {
+          read: () => { return 'Error message in body that gets printed to console.'; }
+        }
+      };
+      spyOn(blockchainClient as any, 'fetch').and.returnValue(Promise.resolve(mockFetchResponse));
+      spyOn(ReadableStream, 'readAll').and.returnValue(Promise.resolve(JSON.stringify({
+        code: 'unused'
+      })));
+
+      try {
+        await blockchainClient.read(1, 'Unused transaction hash.');
+      } catch {
+        // Throwing error is the expected case.
+        return;
+      }
+
+      fail();
+    });
+
+    it('should throw if not both the transaction time and hash are given.', async () => {
+      const blockchainClient = new Blockchain('Unused URI');
+      try {
+        await blockchainClient.read(1, undefined);
+      } catch {
+        // Throwing error is the expected case.
```

Comment:
option to assert more about the nature of the error.

Reply:
Took your suggestion, checking for specific error code.